app/controllers/cart_controller.rb

```diff
@@ -0,0 +1,54 @@
+class CartController < ApplicationController
+  before_action :load_cart
+  before_action :logged_in_user
+
+  def create
+    @product = Product.find_by id: params[:id]
+    quantity = params[:quantity].to_i
+    current_cart_detail = @cart.cart_details.find_by(product_id: @product.id)
+    if current_cart_detail && quantity.positive?
+      new_quantity = current_cart_detail.quantity + quantity
+      current_cart_detail.update(quantity: new_quantity)
+    elsif quantity <= 0
+      handle_add_failed
+      return
+    else
+      @cart.cart_details.create(product: @product, quantity:)
+    end
+    handle_add_success
+  end
+
+  def check_remain_and_redirect
```

Comment:
cái này có phải là .... method new ko?

Reply:
dạ method này em check ở nút "Thanh toán" bên trang giỏ hàng để chuyển sang trang nhập thông tin cho hóa đơn, nếu số lượng sản phẩm trong giỏ hàng lớn hơn số lượng tồn kho thì không cho redirect, còn method new bill là bạn khác làm ạ